### PR TITLE
feat: Add more functionality to template processor

### PR DESCRIPTION
**Pull request by Hipska**
### Required for all PRs:

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

I was surprised the template processor didn't support `{{.}}` in a usable way.
```diff
- {0xc0000565a0}
+ test1 map[tag1:value1] map[value:1] 1257894000000000000
```

In the meantime, I also added support for `{{ .FieldList }}` and `{{ .TagList }}`.

### CODE REVIEW

**Review by srebhan — APPROVED** [review]
Excellent work @Hipska! Looks good to me.

File: plugins/processors/template/README.md [hunk 1]
```diff
@@ -59,4 +59,32 @@ Add the year as a tag, similar to the date processor:
...
+[[processors.template]]
+  tag = "message"
+  template = 'Fields: {{ .FieldList }}'
+  # Advanced example:
+  # template 'Fields: {{ range $name, $value := .FieldList }}{{$name}}:{{$value}} {{end}}'
+```
+
+```diff
+- cpu,hostname=localhost time_idle=42
++ cpu,hostname=localhost,message=Fields:\ map[time_idle:42] time_idle=42
```
Comment (powersj) [inline plugins/processors/template/README.md:74]: Is the message value here interpreted as a string by line protocol?

Comment (Hipska) [reply plugins/processors/template/README.md:74]: Tags are always only strings. I'm just pasting what the line protocol serialiser is outputting

Comment (powersj) [reply plugins/processors/template/README.md:74]: Is there a different example that we can use then? This particular example has me concerned for two reasons:

1. Having a string with what is Golang map's referenced in them seems not useful at all. Is there a use-case I'm missing?
2. If someone doesn't post-process this it is going to blow up their cardinality

Comment (Hipska) [reply plugins/processors/template/README.md:74]: 1. All output plugins that expect a `message` tag or field (syslog, groundwork, ..) where you want to list all metric fields for example. Another example could also be by doing something like `{{ range $field, $value := .FieldList }}`, but I thought that would be too advanced as example?
2. Sure, but InfluxDB is not the only type of output available..

Comment (powersj) [reply plugins/processors/template/README.md:74]: > 1. All output plugins that expect a `message` tag or field (syslog, groundwork, ..) where you want to list all metric fields for example. See also my commented out example, which doesn't include 'map'.
> 

I still do not understand what the use case is for that.

>     2. Sure, but InfluxDB is not the only type of output available.. (While thinking of it, wouldn't it be nice to also have option to directly put result to a field? That would make more sense in the examples)

I understand that we have more than InfluxDB, however, we know users rely on these examples and you cannot be certain a user of InfluxDB starts using your example and then complains when their cardinality blows up.

File: plugins/processors/template/README.md [hunk 2]
```diff
@@ -59,4 +59,32 @@ Add the year as a tag, similar to the date processor:
...
 ```
 
+Add all fields as a tag:
+
+```toml
+[[processors.template]]
+  tag = "message"
+  template = 'Fields: {{ .FieldList }}'
+  # Advanced example:
+  # template 'Fields: {{ range $name, $value := .FieldList }}{{$name}}:{{$value}} {{end}}'
```
Comment (powersj) [inline plugins/processors/template/README.md:69]: I would prefer you remove the simple example and instead uncomment and show the more advanced example here. I think that is more likely what user is going to want versus the "map[.]" text.

**Comment by Hipska** [comment]
@powersj something like this?

**Review by powersj — APPROVED** [review]
Thank you!

**Pull request merged**